frontend/src/components/widgets/DataFrame/DataFrameContainer.tsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license
+ * Copyright 2018-2022 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+
+import styled, { ThemeProvider } from "styled-components"
+import { useTheme } from "@emotion/react"
+import { transparentize } from "color2k"
+import { Theme as GlideTheme } from "@glideapps/glide-data-grid"
+
+import { Theme } from "src/theme"
+
+/**
+ * Creates a glide-data-grid compatible theme based on our theme configuration.
+ *
+ * @param theme: Our theme configuration.
+ *
+ * @return a glide-data-grid compatible theme.
+ */
+export function createDataFrameTheme(theme: Theme): GlideTheme {
+  return {
+    // Explanations: https://github.com/glideapps/glide-data-grid/blob/main/packages/core/API.md#theme
+    accentColor: theme.colors.primary,
+    accentFg: theme.colors.white,
+    accentLight: transparentize(theme.colors.primary, 0.9),
+    borderColor: theme.colors.fadedText05,
+    fontFamily: theme.genericFonts.bodyFont,
+    bgSearchResult: transparentize(theme.colors.primary, 0.9),
+    // Header styling:
+    bgIconHeader: theme.colors.fadedText60,
+    fgIconHeader: theme.colors.white,
+    bgHeader: theme.colors.bgMix,
+    bgHeaderHasFocus: theme.colors.secondaryBg,
+    bgHeaderHovered: theme.colors.bgMix, // uses same color as bgHeader to deactivate the hover effect
+    textHeader: theme.colors.fadedText60,
+    textHeaderSelected: theme.colors.white,
+    headerFontStyle: `${theme.fontSizes.sm}`,
```

Comment:
nit: Is there a reason why this is a string and not just `theme.fontSizes.sm`

Reply:
I think the reason here is that the style has to be a string and not a number. Therefore, I'm using a template string here. A `.toString()` might work as well here.